src/m365/spo/commands/site/site-admin-list.ts

```diff
@@ -0,0 +1,262 @@
+  Id?: number;
+  Email: string;
+  IsPrimaryAdmin: boolean;
+  LoginName: string;
+  Title: string;
+  PrincipalType?: number;
+  PrincipalTypeString?: string;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  asAdmin?: boolean;
+}
+
+class SpoSiteAdminListCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_ADMIN_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all administrators of a specific SharePoint site';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        siteUrl: typeof args.options.siteUrl !== 'undefined',
+        asAdmin: typeof args.options.asAdmin !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --siteUrl <siteUrl>'
+      },
+      {
+        option: '--asAdmin'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => validation.isValidSharePointUrl(args.options.siteUrl)
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      if (args.options.asAdmin) {
+        await this.callActionAsAdmin(logger, args);
+        return;
+      }
+
+      await this.callAction(logger, args);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async callActionAsAdmin(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      await logger.logToStderr('Retrieving site administrators as an administrator...');
+      const adminUrl: string = await spo.getSpoAdminUrl(logger, this.debug);
+      const siteId = await this.getSiteIdBasedOnUrl(logger, args.options.siteUrl, adminUrl);
+
+      const requestOptions: CliRequestOptions = {
+        url: `${adminUrl}/_api/SPO.Tenant/GetSiteAdministrators?siteId=%27${siteId}%27`,
+        headers: {
+          accept: 'application/json;odata=nometadata',
+          'content-type': 'application/json;charset=utf-8'
+        }
+      };
+
+      const response: string = await request.post<string>(requestOptions);
+      const responseContent: AdminResult = JSON.parse(response);
+      const primaryAdminLoginName = await this.getPrimaryAdminLoginNameFromAdmin(adminUrl, siteId);
+
+      const mappedResult = responseContent.value.map((u: AdminUserResult): CommandResultItem => {
+        return {
+          Id: undefined,
+          Email: u.email,
+          LoginName: u.loginName,
+          Title: u.name,
+          PrincipalType: undefined,
+          PrincipalTypeString: undefined,
+          IsPrimaryAdmin: u.loginName === primaryAdminLoginName
+        };
+      });
+      await logger.log(mappedResult);
+    }
+    catch (err: any) {
+      throw err;
+    }
```

Comment:
if we don't throw any custom error or don't add any additional error handling here, it makes little sense to have an 'empty' `try... catch` just to `throw` and `catch` in the parent method. We may skip this `try.. catch` block and just do the error handling in the parent method

Reply:
Done